### PR TITLE
[Woptim] Update Shared CI and Spack Configs

### PR DESCRIPTION
**Pull request by adrienbernede**
# Summary

- This PR updates the Shared CI
- This PR updates the submodule RADIUSS Spack Configs

## Purpose

In Shared CI:
- Fix reproducer to print needed environment variables

In RADIUSS Spack Configs:
- The local CARE package has been removed, now using the one in RADIUSS Spack Configs.
- The compilers (wrappers) on Lassen have been updated to better support the use of gcc 8.3.1 toolchain.
- The local spack packages have been updated to be as close as possible to what we are pushing in https://github.com/spack/spack/pull/41375.
- We moved to Spack environments.

## Notes

Packages in RADIUSS Spack Configs still have some specific changes compared to the PR mentioned above.

### CODE REVIEW

**Comment by adrienbernede** [comment]
@adayton1 this is ready for review !

File: scripts/spack_packages/repo.yaml [hunk 1]
```diff
@@ -1,2 +0,0 @@
-repo:
```
Comment (adayton1) [inline scripts/spack_packages/repo.yaml:1]: I'd rather this machinery be left in.

File: .uberenv_config.json [hunk 2]
```diff
@@ -4,10 +4,10 @@
 "package_final_phase": "initconfig",
 "package_source_dir": "../..",
 "spack_url": "https://github.com/spack/spack.git",
-"spack_branch": "develop-2024-01-21",
+"spack_branch": "develop-2024-02-18",
 "spack_activate": {},
 "spack_configs_path": "scripts/radiuss-spack-configs",
-"spack_packages_path": ["scripts/radiuss-spack-configs/packages", "scripts/spack_packages/packages"],
```
Comment (adayton1) [inline .uberenv_config.json:10]: I'd rather have this machinery left in.

Comment (adrienbernede) [reply .uberenv_config.json:10]: Done!

File: scripts/spack_packages/repo.yaml (hunk 1 above)
Comment (adrienbernede) [reply scripts/spack_packages/repo.yaml]: Done!

**Comment by adrienbernede** [comment]
@adayton1 sorry for the delay... I lost sight of it.

**Comment by adayton1** [comment]
> @adayton1 sorry for the delay... I lost sight of it.

No worries! Is this one completely superseded by #263 ?

**Comment by adrienbernede** [comment]
Closing, superseded by #263.
(@adayton1 I forgot to get this one first, but #263 had the full history, so we are good).

**Pull request closed without merging**